lib/ecto/schema.ex

```diff
@@ -221,6 +224,8 @@ defmodule Ecto.Schema do
   used for runtime introspection of the schema:
 
   * `__schema__(:source)` - Returns the source as given to `schema/2`;
+  * `__schema__(:prefix)` - Returns optional prefix for source provided by
```

Comment:
Instead of `__schema__(:prefix)`, I believe we should make `__schema__(:source)` return a tuple. Then we check everywhere we call `__schema__(:source)` to verify it still makes sense.